components/esp_lcd_touch_gt911/esp_lcd_touch_gt911.c

```diff
@@ -250,31 +246,19 @@ static void touch_gt911_read_cfg(esp_lcd_touch_handle_t tp)
 
 static esp_err_t touch_gt911_i2c_read(esp_lcd_touch_handle_t tp, uint16_t reg, uint8_t *data, uint8_t len)
 {
-    esp_lcd_touch_dev_i2c_t *gt911_dev = NULL;
-    uint8_t reg_buf[2] = {(uint8_t)((reg >> 8) & 0xFF), (uint8_t)(reg & 0xFF)};
-
     assert(tp != NULL);
     assert(data != NULL);
 
-    gt911_dev = &tp->config.device.i2c;
-    assert(gt911_dev->port >= 0 && gt911_dev->port < SOC_I2C_NUM);
-
-    /* Write register number */
-    ESP_RETURN_ON_ERROR(i2c_master_write_to_device(gt911_dev->port, ESP_LCD_TOUCH_GT911_I2C_ADDRESS, reg_buf, sizeof(reg_buf), ESP_LCD_TOUCH_GT911_I2C_TIMEOUT_MS / portTICK_PERIOD_MS), TAG, "I2C write error!");
-
     /* Read data */
-    return i2c_master_read_from_device(gt911_dev->port, ESP_LCD_TOUCH_GT911_I2C_ADDRESS, data, len, ESP_LCD_TOUCH_GT911_I2C_TIMEOUT_MS / portTICK_PERIOD_MS);
+    return esp_lcd_panel_io_rx_param(tp->io, reg, data, len);
 }
 
 static esp_err_t touch_gt911_i2c_write(esp_lcd_touch_handle_t tp, uint16_t reg, uint8_t data)
 {
-    esp_lcd_touch_dev_i2c_t *gt911_dev = NULL;
-    uint8_t reg_buf[3] = {(uint8_t)((reg >> 8) & 0xFF), (uint8_t)(reg & 0xFF), data};
-
     assert(tp != NULL);
 
-    gt911_dev = &tp->config.device.i2c;
-    assert(gt911_dev->port >= 0 && gt911_dev->port < SOC_I2C_NUM);
-
-    return i2c_master_write_to_device(gt911_dev->port, ESP_LCD_TOUCH_GT911_I2C_ADDRESS, reg_buf, sizeof(reg_buf), ESP_LCD_TOUCH_GT911_I2C_TIMEOUT_MS / portTICK_PERIOD_MS);
+    // *INDENT-OFF*
+    /* Write data */
+    return esp_lcd_panel_io_tx_param(tp->io, reg, (uint8_t[]){data}, 1);
+    // *INDENT-ON*
```

Comment:
Likewise: could single-line formatting disable used?

Reply:
This is not working...